### PR TITLE
Add r-arrow, R Integration to Apache Arrow

### PR DESCRIPTION
**Pull request by dbast**
Helps to speedup sparklyr 1.0 https://blog.rstudio.com/2019/03/15/sparklyr-1-0/

Pinging @conda-forge/help-r and @conda-forge/arrow-cpp and @conda-forge/pyarrow for review / interest in co-maintenance.

Feedback welcome :)

### CODE REVIEW

File: recipes/r-arrow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+about:
+  home: https://github.com/apache/arrow
+  license: Apache 2.0
+  summary: R Integration to 'Apache' 'Arrow'.
+  license_family: APACHE
+
+extra:
+  recipe-maintainers:
+    - conda-forge/r
```
Comment (xhochy) [inline recipes/r-arrow/meta.yaml]: Feel free to add @conda-forge/arrow-cpp as a maintainer, we can then update this package with all other Arrow packages when we make a release.

File: recipes/r-arrow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  skip: True  # [win]
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+requirements:
+  build:
+    - {{ compiler('c') }}        # [not win]
+    - {{ compiler('cxx') }}      # [not win]
+    - {{posix}}pkg-config
```
Comment (xhochy) [inline recipes/r-arrow/meta.yaml]: We can probably get rid nowdays of the `{{posix}}` prefix here as https://github.com/conda-forge/pkg-config-feedstock is now also built for Windows. (related to this feedstock).

File: recipes/r-arrow/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  version: {{ version|replace("-", "_") }}
+
+source:
+  url: https://dist.apache.org/repos/dist/release/arrow/arrow-{{ version }}/apache-arrow-{{ version }}.tar.gz
+  sha256: e93e43343544e344bbc912b89da01d8abf66596f029d26b2b135b102a9e39895
+
+build:
+  merge_build_host: True  # [win]
+  number: 0
+  skip: True  # [win]
```
Comment (xhochy) [inline recipes/r-arrow/meta.yaml]: What are the problems this is skipped on Win?

File: recipes/r-arrow/meta.yaml (hunk 1 above)
Comment (dbast) [reply recipes/r-arrow/meta.yaml]: ok. done

File: recipes/r-arrow/meta.yaml (hunk 2 above)
Comment (dbast) [reply recipes/r-arrow/meta.yaml]: ok. done

File: recipes/r-arrow/meta.yaml (hunk 3 above)
Comment (dbast) [reply recipes/r-arrow/meta.yaml]: @xhochy Good question: Tried to build on windows 2-3 times before opening the PR, which was not successful, see e.g. https://ci.appveyor.com/project/dbast/staged-recipes/builds/23117017

Then I had a look into the upstream repo and found no indication that this was ever build on windows (./r excluded in appveyor.yml ... nothing windows related in ./r/src/configure and ./r/README.md, all about macOS / various Linux OSs... no ./r/src/Makewars.win etc.) -> Decided to skip for now until there is something upstream ... 

So we can keep the skip for now and activate windows later in the feedstock or do it here, if there is some confidence that this works with the 0.12.1 src tarball ...

Comment (xhochy) [reply recipes/r-arrow/meta.yaml]: Oh, sorry. Forgot that we're still missing this upstream

Comment (dbast) [reply recipes/r-arrow/meta.yaml]: ok. thanks for reviewing and confirming!

**Pull request merged**